### ISSUE TITLE
Problem with small value for 'window' in distance_matrix_fast()

### ISSUE THREAD

**Issue by asyates** [wannesm/dtaidistance#131]
Hi,

Just another thing I stumbled on. I am using the distance_matrix_fast() function to compute the distance between many different waveforms, and then I'm performing hierachical clustering (using scipy tools for now).

When setting my window length to a low value I run into one of two problems. With arrays of length 500 being compared, setting the window size to 10-15 results in distances of 0 (or some very small float) being returned (this is definitely not the case). Below are the dendrograms showing a window=15 and a window=20, so distance on the y axis. Minor change in window length relative to the array length... but one is returning ~0 distances (window=15) and the other the expected behaviour (window=20).

![window15](https://user-images.githubusercontent.com/50132148/125082817-f21c8080-e0c7-11eb-85e6-5a86ae5f2fa4.png)
![window20](https://user-images.githubusercontent.com/50132148/125082823-f47eda80-e0c7-11eb-86fb-6c4f0b8ecb4f.png)

At lower values, e.g. trying to set window=1 for the euclidean distance, I get an errors:

```
malloc(): invalid size (unsorted)
malloc(): invalid size (unsorted)
malloc(): invalid size (unsorted)
malloc(): invalid size (unsorted)
malloc(): invalid size (unsorted)
```
etc. 

Any idea what causes this behaviour?

**Comment by asyates** [comment]
Reproducing similar behaviour using function below. Might be related to other issue I raised as only seems to happen when psi parameter is defined (#130) or perhaps it's just a limitation of the algorithm I'm not aware of based on relative values of window and psi (relatively new to DTW).

```
def test03(win=None ,psi=None):

    x = np.arange(0, 250, .5)
    s1 = np.sin(x)
    s2 = np.sin(x-1)

    random.seed(1)
    for idx in range(len(s2)):
        if random.random() < 0.05:
            s2[idx] += (random.random() - 0.5) / 2

    d, paths = dtw.warping_paths(s1, s2, psi=psi, window=win)
    best_path = dtw.best_path(paths)

    dtwvis.plot_warpingpaths(s1, s2, paths, best_path)
    dist = dtw.distance_fast(s1, s2, psi=psi, window=win)

    print('array length = '+str(len(s1))+', window length = '+str(win))
    print('psi = '+str(psi))
    print('returned distance = '+str(dist))

    plt.show()
```

So for example, test03(win=20, psi=None) returns:

```
In [2]: test03(win=20, psi=None)
array length = 500, window length = 20
psi = None
returned distance = 1.4975831197795406
```
![psi_win_test0](https://user-images.githubusercontent.com/50132148/125091197-80950000-e0d0-11eb-8530-e08d3707efb3.png)

so distances matching.

However, test03(win=20, psi=50) returns:
```
In [7]: test03(win=20, psi=50)
array length = 500, window length = 20
psi = 50
returned distance = 0.0
```
![psi_win_test1](https://user-images.githubusercontent.com/50132148/125091426-ade1ae00-e0d0-11eb-9d28-36dbd8253137.png)

so results aren't matching once psi parameter is introduced (with distance_fast returning 0 value).

Also, the values never seem to match when setting the psi, which again might be related to previous issue I raised #130 e.g.
```
In [12]: test03(win=5, psi=10)
array length = 500, window length = 5
psi = 10
returned distance = 0.0

In [13]: test03(win=5, psi=1)
array length = 500, window length = 5
psi = 1
returned distance = 1.2388227404980305
```
whereas the distance indicated by the warping path plot was 0.8347 (compared to 0) and 1.0495 (compared to 1.238..) respectively.

Is there something I am missing, or is this a bug?

**Comment by wannesm** [comment]
There’s indeed a bug in what value in the warping paths matrix that is returned (in a particular case). Good call. Should be fixed in a few days.

**Comment by wannesm** [comment]
The problem with psi-relaxation in the second post is fixed in master in commit 82033215e2f8dc7e2659e537b55a821946679c28 .
